lib/spack/spack/patch.py

```diff
@@ -329,17 +330,20 @@ class PatchCache(object):
 
     """
 
-    def __init__(self, data=None):
+    def __init__(self, data=None, repository=None):
         if data is None:
             self.index = {}
         else:
             if "patches" not in data:
                 raise IndexError("invalid patch index; try `spack clean -m`")
             self.index = data["patches"]
 
+        assert repository is not None, "a 'repository=' argument is required"
```

Comment:
The idea was to avoid changing the signature from:
```python
def __init__(self, data=None):
```
to:
```python
def __init__(self, repository, data=None):
```
but I can do that if it is preferable.